### ISSUE TITLE
hangs on preset  click

### ISSUE THREAD

**Issue by gilles504** [LMMS/lmms#2315]
With latest master branch (up to date) I experience frequent hangs on a linux 32 bit host (ubuntu 12.04.2-LTS alsa).
Preference directory was cleaned. Lmms was cloned from git yesterday and compiled without error.
Seems stable otherwise (playing, recording midi). But hangs randomly when clicking on a preset.
no message on the console.

Is there a method to disable audio preview of a preset ?
```
[more info]
seems a thread exits when the hang happens
Using host libthread_db library "/lib/i386-linux-gnu/libthread_db.so.1".
[New Thread 0xaf833b40 (LWP 13603)]
[New Thread 0xaeeffb40 (LWP 13604)]
[New Thread 0xae4ffb40 (LWP 13605)]
[New Thread 0xaceeab40 (LWP 13606)]
[New Thread 0xac57eb40 (LWP 13607)]
[New Thread 0xabd7db40 (LWP 13608)]
[New Thread 0xab3e4b40 (LWP 13609)]
[New Thread 0xaabe3b40 (LWP 13610)]
mouse press -> Add play
[Thread 0xae4ffb40 (LWP 13605) exited]
```
If I remove call to `if( !Engine::mixer()->addPlayHandle` in `FileBrowser.cpp` the hang disapears (and preview should also... but strangly not with presets for the Opulenz plugin... (LB302 plugin preset also makes some noise... but not the preset itself...)

**Comment by Umcaruje** [comment]
Hi @gilles504 could you please provide us with a specific preset that creates this crash?

**Comment by tresf** [comment]
Duplicate of #2269 perhaps?

**Comment by gilles504** [comment]
just got the hang on SID lead.xpf preset. happens if I select many preset in a short time.

**Comment by Umcaruje** [comment]
confirmed on Ubuntu 15.04 x64 - here is a backtrace: https://gist.github.com/Umcaruje/618c37c22ff76516a482

How to reproduce: Click several sid presets quickly one after another and lmms will hang.

**Comment by musikBear** [comment]
I would like to suggest a heuristic way to steer *around* this critical bug.
My suggestion is to completely remove **all** response to left-mouse and only allow presets, in the sidebar, to be right-clicked.
A right-click should then show a new item in the context-menu
**"Listen to this preset"**
This should be the *only* possible way, to activate preview.
The user will now not have the opportunity  to click too fast on various presets, and that would protect LMMS from crashing, for *that* reason.
A necessary feature is needed. 
LMMS need *not* to show this new context-meny item at all, *if* the preset is of type VST.
@curlymorphic & code-wizards:  Is that possible??
If so, the other critical bug, where LMMS opens a VST gui, is also solved, and It would reduce the number of criticals to 3..

**Comment by tresf** [comment]
@musicbear, anything is possible.  This won't happen, sorry.

**Comment by musikBear** [comment]
@tresf  *sigh* ok.. I just felt a different angle to this problem, could be beneficial -but, are the criticals being worked at? I miss some kind of discussion on what to do now? 
What is the plan, forward?

**Comment by BaraMGB** [comment]
Can't reproduce this on Ubuntu 14.04 64bit

**Comment by tresf** [comment]
@BaraMGB thanks for testing.

I was on 12.04 64-bit using latest master reproducing this in November here: https://github.com/LMMS/lmms/issues/2378#issuecomment-154121997.  AFAIK, this still exists.

**Comment by Fastigium** [comment]
Alright, I was able to reproduce this bug on 14.04.3 64-bit and managed to fix it by altering the code as submitted in https://github.com/LMMS/lmms/pull/2497. I'd be grateful for any testing to confirm or deny that this fixes the issue. However, the code quality of the fixes in the pull request is sub-par IMO. I'd like to go over them with someone more at home in the code base.

**Comment by Fastigium** [comment]
Hey, can I get a bit more response on #2497? I've come part of the way fixing this, but I need some input on how to proceed. Thanks!

**Comment by musikBear** [comment]
@tresf Is this one not fixed by @Fastigium ? Your build worked perfect for me.
If it is fixed, theres only one critical left. And after **that one** is crushed, then imo a lot (all)of remaining issues, could be re-scheduled to 1.2.1

**Comment by tresf** [comment]
> @tresf Is this one not fixed by @Fastigium ? Your build worked perfect for me.

He's waiting for advice over at his PR.

> after that one is crushed, then imo a lot (all)of remaining issues, could be re-scheduled to 1.2.1

I think what you're trying to say is we can make an RC after this is merged.  If that's the case, I agree.

**Comment by musikBear** [comment]
Let me know if tests need to be done on @Fastigium latest changes

**Comment by Fastigium** [comment]
My biggest issue with merging my PR #2497 is lack of feedback on the code. I don't feel confident about my style, but there doesn't appear to be anyone atm to tell me if I'm doing it right and/or how to improve. Testing-wise, I think it's RC-ready. The computer that generated the mysterious crashes actually died a week ago, and my other testing machine and @musikBear's were crash-free.

**Comment by musikBear** [comment]
> and my other testing machine and @musikBear's were crash-free.

@Fastigium @tresf 

Better actually. I have used your PR as if it was a beta, and in respect to stability during open usage, its good!
It does however 'crash'  every time at shut-down, just as Master have done since late summer, eg #2088

**Comment by tresf** [comment]
Closed via #2315.
